plugins/hls-semantic-tokens-plugin/src/Ide/Plugin/SemanticTokens/Utils.hs

```diff
@@ -119,4 +123,18 @@ recoverFunMaskArray flattened = unflattened
     go (HTyConApp _ _)             = False
 
 
+realSrcSpanToCodePointRange :: RealSrcSpan -> CodePointRange
```

Comment:
let's put these somewhere more central, like in `hls-plugin-api`. Ideally more people would start to use these!

Reply:
Yep ,these two are meant to be a replacement for the  `realSrcSpanToRange , realSrcLocToPosition`, I think we should put the codePoint version beside these two, and deprecate them(ideally `Range, Position` should be in lsp standard, `CodePointRange, CodePointPosition` should be in ghc standard). 

In the mean time, I think the lsp package could cache the ghc position to lsp position mapping in the virtual file type, making the conversion to be more efficient. Hopefully, every other part of the hls could support unicode eventually.

Reply:
I goes through the status, the lsp3.17 protocal does support codePoint, but need to wait for the client support, up to today, vscode still does not support https://github.com/microsoft/vscode-languageserver-node/blob/3f725d8368ec0c75f390a37e4e159d78bc1527ac/client/src/common/client.ts#L1240, :(